### DISCUSSION TITLE
Keyboard input freezes

### DISCUSSION THREAD

**Discussion by mahescho** [rustdesk/rustdesk#8791] [Q&A] [answered]
Every now and them keyboard input is no longer possible in sessions. Mouse still works. After closing and reestablishing the session it works again.

Any suggestions how to fix this?

(RustDesk Server Pro 1.4.0)

**Comment by rustdesk** [comment]
similar to this, https://github.com/rustdesk/rustdesk/issues/8768? please tell the os on each side. This has nothing to do with RustDesk server.

**Reply by mahescho** [reply]
Perhaps. My problem is a bit different. Mouse movement does not fix the typing problem. Mouse works, typing does not.

Control OS is Linux and I've observed this with Mac OS and also Windows Clients.

**Reply by dignow** [reply]
It seems a bug of keyboard grabbing on Linux.

Which distro and version do you use?
Do you use x11 on the control side?

**Reply by rustdesk** [reply]
Controlling (local) side is Linux?

**Reply by rustdesk** [reply]
How about `input source 2`?

![image](https://github.com/user-attachments/assets/fe35ced3-4cd7-427e-a4e3-ac2d49602ae1)

**Reply by mahescho** [reply]
Contolling (loacal) side is Linux, KDE Neon with Wayland.

When connected to a Mac my Keyboard settings look like this:

![image](https://github.com/user-attachments/assets/290c1a51-ede3-4f07-858a-c1f02ea36322)

When connected to Windows:

![image](https://github.com/user-attachments/assets/8162a482-5f9f-45d5-92aa-4eb08118b81d)

When connected to Linux:

![image](https://github.com/user-attachments/assets/4d073ddd-8408-48bf-9786-8e2632a404a1)

Client version: Version: 1.2.7

**Reply by rustdesk** [reply]
you could try out input source 2

![image](https://github.com/user-attachments/assets/928d984e-73f4-408b-9c15-1b71b5addda5)

**Reply by mahescho** [reply]
Well, as you can see in my screenshots: There is no "input source 2".

**Reply by fufesou** [reply]
@mahescho Hi, which language do you use on the local side?

Please also confirm is it stop working after connection
1. For a while with no use
2. Or you have to keep using it.

I've installed `Kubuntu 24.04` and `neon-user-20240725-0828.iso`, both english input, and within 30 minutes it all worked fine.
I'll continue testing.

**Reply by mahescho** [reply]
My user uses German localisation and RustDesk is set to English manually, just for posting messages and screenshots here and for easier documentation lookups :-)

I left my PC at 17:00 with a working connection to the Mac. I  returned 20:15 and no keyboard input possible. BTW: Mean while my Linux screen got locked by inactivity timeout. The connected Mac does not get locked.

So it's "For a while with no use".

**Reply by fufesou** [reply]
Thanks for your feedback.

I've kept my connections `Kubuntu 24.04` and `neon-user-20240725-0828.iso` -> `MacOS` for about 9 hours. The keyboard input still works fine.

I'm not sure if it is an issue of 
1. keyboard listening
2. key events process (Conversion and transportation)
3. or key events simulation

in your connections.

Could you please also try X11 (on local side) with "Input source 1"? If the issue is gone, it means the keyboard listening is not stable in Wayland. Thx

**Reply by mahescho** [reply]
Hmmm ... one more thing. My monitor features USB ports and a KVM switch. I use it to switch between my Linux and Mac. When I leave I turn it off. So keyboard an mouse get disconnected. I'm not sure if there is any conjunction with this.

I will try different variants, observe the behavior and then report back.

**Reply by fufesou** [reply]
> Maybe this is the difference.

Although I also think that the keyboard should work again. But I tend to think this is a flutter issue.

**Reply by mahescho** [reply]
Ah .. flutter ... now I know why window decorations are missing and the complete window management functionality of Plasma is missing or better: "is broken" :-) I really miss the Plasma features.

**Comment by mahescho** [comment]
Well this problem is getting worse. Now Keyboard stops working while in a active session using the mous.

I really need a fix for this!

**Reply by fufesou** [reply]
Does the mouse always work? Do you have any different settings on your mouse?
The most likely place for this issue is the control side. Or there should be a lot of reports about this issue.
But mouse events and keyboard events are handled separately in our code.
Then, there may be something wrong in low level processing (eg. flutter, hardware drivers).

Could you please also try X11 on the control side?

**Reply by mahescho** [reply]
The mouse works fails. The only correlation I've found is the up time of my Linux desktop. I rarely reboot it. But 2 days ago I did. At first I thought the problem vanished, but today the Mac session lost keyboard input. Perhaps there is a correlation with the up time and no coincidence ...

A X11 setup is possible but I've to do a fresh install in a VM. I've no active X11 setup at my (control) site.

**Reply by mahescho** [reply]
For testing and mainly because of "Send clipboard keystroke (https://github.com/rustdesk/rustdesk/discussions/5451)" I switched to nightly for control.

**Comment by mahescho** [comment]
New observations:

I had a long running Mac OS session without any problems. Then I had to start a new, parallel Window session. As soon as I've started the new Windows session the keyboard input in the Mac OS session stopped working. This Window session perfectly worked for more than 12 hours. Now I've started an other Windows session and the keyboard input of the previously working Windows session stopped working. All the same with a new Linux session.

My conclusion:

The keyboard input only works for latest Session with control on Linux Wayland.

**Reply by mahescho** [reply]
Matter of fact: only the latest session gets keyboard input.

**Reply by fufesou** [reply]
> Matter of fact: only the latest session gets keyboard input.

Thank you very much for figuring out the fact, I can reproduce.

**Comment by fufesou** [comment, marked as answer]
https://github.com/rustdesk/rustdesk/pull/9189 may fix this issue.

**Reply by mahescho** [reply]
I think you are right! I will test the nightly tomorrow.

I don't use tabs.

![image](https://github.com/user-attachments/assets/544f8da7-0047-4165-a50c-f5b4bb1d7de4)

Every session is a new window. When I click in the window mouse works but keyboard does not get the focus. As soon as I  klick on the (singe!) tab of the window keyboard starts working again. Till now I never clicked the tab ... because I use windows ...

**Reply by mahescho** [reply]
Ok, tested the current nightly. Works as expected.

**Reply by MatthewZMD** [reply]
I am using Nightly, it was working for a day, but broken again

**Reply by fufesou** [reply]
@MatthewZMD Hi, thanks for your feedback.

What're the OSs on both sides? And what input source do you use? If the control side is Linux, is it X11 or Wayland?

![a](https://github.com/user-attachments/assets/0f9d5290-eebd-4bd1-bde3-be312d59ee90)

Please give more details or reproduce steps if possible, thanks.

**Reply by rustdesk** [reply]
> I am using Nightly, it was working for a day, but broken again

Try out today's nightly, there is bug in old nightly. https://github.com/rustdesk/rustdesk/issues/9263#issuecomment-2340296203

**Reply by MatthewZMD** [reply]
@fufesou Thanks for getting back quickly!

I'm controlling a Linux Mint (X11) from a Macbook M3, the control side is Macbook and using the `Map mode`.

I cannot find a consistently reproducible step for the problem, it was happening occasionally for several months and quite frustrating. Thanks for tackling on this problem, I'm trying the new nightly and it has been working again so far!

I will keep you posted if it ever break again.

**Reply by MatthewZMD** [reply]
FYI encountering same issue again with the same setup. Restarting Rustdesk from the Macbook resolved it.
